src/features/generic_events/aggregate/index.js

```diff
@@ -118,6 +135,9 @@ export class Aggregate extends AggregateBase {
   }
 
   onHarvestStarted (options) {
+    /** send whatever UserActions have been aggregated up to this point
+     * if we are in a final harvest */
+    if (options.isFinalHarvest) this.userActionAggregator?.storeCurrentUserActionInFeature()
```

Comment:
double checking: if finalHarvest **and** there's a pending UA, call `this.addEvent` on it -- might have to re-write this with the other changes